src/compiler/checker.ts

```diff
@@ -3603,6 +3604,19 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         );
                     }
                 }
+
+                // Look at 'compilerOptions.isolatedModules' and not 'getIsolatedModules(...)' (which considers 'verbatimModuleSyntax')
+                // here because 'verbatimModuleSyntax' will already have an error for importing a type without 'import type'.
+                if (compilerOptions.isolatedModules && result && isInExternalModule && meaning & SymbolFlags.Value) {
+                    const isGlobal = lookup(globals, name, meaning) === result;
+                    const typeSymbol = isSourceFile(lastLocation!) && lastLocation.locals && lookup(lastLocation.locals, name, SymbolFlags.Type);

```

Comment:
I don’t think this is quite right—the characteristic of the local symbol that makes it problematic isn’t that it _is a type_; it’s that it _is not a value_. In other words, there could be potential for false positives or false negatives in a few cases here:

- the imported symbol is a type and a value at the same time, like a class (I don’t think this case is possible since `isGlobal` would be false)
- the imported symbol is an uninstantiated namespace
- the imported symbol was unresolvable and so has all `SymbolFlags` (again, shouldn’t be an issue here since `isGlobal` would be false)

Of these corner cases to think about, it sounds like uninstantiated namespaces might pose a problem. Here’s a test:

```ts
// @isolatedModules: true
// @noEmit: true
// @noTypesAndSymbols: true

// @filename: ./types.ts
export namespace Date {
  export type T = any;
}

// @filename: ./bad.ts
import { Date } from './types';
new Date();
```

Maybe a lookup with `~SymbolFlags.Value` instead of `SymbolFlags.Type` would better capture the intent?

Reply:
Good catch and thanks for the explanation. Changed to `~SymbolFlags.Value` and added the test case.